frontend/src/components/referenceGenomes/EditReferenceGenomes.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useCallback, useState } from "react"
+import AppPageHeader from "../AppPageHeader";
+import PageContent from "../PageContent";
+import { Button, Form, FormItemProps, Input, Select, Space } from "antd";
+import { useNavigate, useParams } from "react-router-dom";
+import { useAppDispatch, useAppSelector } from "../../hooks";
+import { add, list, update } from "../../modules/referenceGenomes/actions";
+import { requiredRules } from "../../constants";
+import { selectAppInitialzed, selectReferenceGenomesByID, selectTaxonsByID } from "../../selectors";
+import * as Options from "../../utils/options";
+import { ReferenceGenome } from "../../models/frontend_models";
+
+interface EditReferenceGenomesProps {
+    referenceGenome?: ReferenceGenome
+}
+
+export const AddReferenceGenomeRoute = () => {
+    const appInitialized = useAppSelector(selectAppInitialzed)
+    return appInitialized ? <EditReferenceGenomes /> : null
+};
+
+export const EditReferenceGenomeRoute = () => {
+    const { id } = useParams()
+    const referenceGenomes = useAppSelector(selectReferenceGenomesByID)
+    const appInitialized = useAppSelector(selectAppInitialzed)
+    return (id && referenceGenomes[id] && appInitialized) ? <EditReferenceGenomes referenceGenome={{ ...referenceGenomes[id] }} /> : null
+}
+
+
+const EditReferenceGenomes = ({ referenceGenome }: EditReferenceGenomesProps) => {
+    const { Item } = Form
+
+    const [form] = Form.useForm()
+    const [formErrors, setFormErrors] = useState({})
+
+    const isAdding = referenceGenome === undefined
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+
+    const taxonsByID = useAppSelector(selectTaxonsByID)
+    const taxonOptions = Object.keys(taxonsByID).map((id) => taxonsByID[id])
```

Comment:
I wrote a function to get the array of items out of an ItemsByID object, `getAllItems` in `frontend_models.ts` so you could use that here.